kura/org.eclipse.kura.container.orchestration.provider/src/main/java/org/eclipse/kura/container/orchestration/provider/impl/enforcement/AllowlistEnforcementMonitor.java

```diff
@@ -0,0 +1,76 @@
+/*******************************************************************************
+ * Copyright (c) 2022, 2024 Eurotech and/or its affiliates and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Eurotech
+ *******************************************************************************/
+
+package org.eclipse.kura.container.orchestration.provider.impl.enforcement;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.eclipse.kura.KuraException;
+import org.eclipse.kura.container.orchestration.provider.impl.ContainerOrchestrationServiceImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.dockerjava.api.async.ResultCallbackTemplate;
+import com.github.dockerjava.api.model.Event;
+
+public class AllowlistEnforcementMonitor extends ResultCallbackTemplate<AllowlistEnforcementMonitor, Event> {
+
+    private static final Logger logger = LoggerFactory.getLogger(AllowlistEnforcementMonitor.class);
+    private static final String ENFORCEMENT_SUCCESS = "Enforcement allowlist contains image digests {}...container {} is starting";
+    private static final String ENFORCEMENT_FAILURE = "Enforcement allowlist doesn't contain image digests...container {} will be stopped";
+    private final List<String> enforcementAllowlistContent;
+    private final ContainerOrchestrationServiceImpl orchestrationServiceImpl;
+
+    public AllowlistEnforcementMonitor(String allowlistContent,
+            ContainerOrchestrationServiceImpl containerOrchestrationService) {
+
+        this.enforcementAllowlistContent = Arrays
+                .asList(allowlistContent.replaceAll("\\s", "").replace("\n", "").trim().split(","));
```

Comment:
I would suggest to split the list on newlines actually.

When creating the list of digests it's quite hard to read them if they are separated by commas, while, separating them by newlines, you have a much easier time reading the different digests.

There's nothing technically wrong in separating on commas, I also think that we use it more often than newlines as separators for multi-parameter field, but in this particular case I think it's much more readable for the user.

```suggestion
        this.enforcementAllowlistContent = Arrays.asList(allowlistContent.split("\\r?\\n|\\r"));
```